contrib/babelfishpg_unit/test_money.c

```diff
@@ -0,0 +1,298 @@
+    bool obtained = DatumGetBool(temp);    
+
+    TEST_ASSERT(expected == obtained, expected?"True":"False", obtained?"True":"False", testResult);
+    return testResult;
+}
+
+
+TestResult*
+test_int4_fixeddecimal_eq(void)
+{
+    Datum val1;
+    Datum val2;
+
+    TestResult* testResult = palloc0(sizeof(TestResult));
+    testResult->result = true;
+
+    val1 = 45;
+    val2 = 94;
+
+    bool expected = false;
+    Datum temp = DirectFunctionCall2(int4_fixeddecimal_eq, val1, val2);
+    bool obtained = DatumGetBool(temp);    
+
+    TEST_ASSERT(expected == obtained, expected?"True":"False", obtained?"True":"False", testResult);
+    return testResult;
+}
+
+
+TestResult*
+test_int4_fixeddecimal_lt(void)
+{
+    Datum val1;
+    Datum val2;
+
+    TestResult* testResult = palloc0(sizeof(TestResult));
+    testResult->result = true;
+
+    val1 = 100;
+    val2 = 10;
+
+    bool expected = false;
+    Datum temp = DirectFunctionCall2(int4_fixeddecimal_lt, val1, val2);
+    bool obtained = DatumGetBool(temp);    
+
+    TEST_ASSERT(expected == obtained, expected?"True":"False", obtained?"True":"False", testResult);
+    return testResult;
+}
+
+
+TestResult*
+test_int4_fixeddecimal_cmp(void)
+{
+    Datum val1;
+    Datum val2;
+    Datum expected;
+    Datum obtained;
+    char expected_str[MAX_TEST_MESSAGE_LENGTH];
+    char obtained_str[MAX_TEST_MESSAGE_LENGTH];
+
+    TestResult* testResult = palloc0(sizeof(TestResult));
+    testResult->result = true;
+
+    val1 = 110;
+    val2 = 100;
+
+    /*
+     * val1 > val2 then result will be 1
+     * val1 == val2 then result will be 0
+     * val1 < val2 then result will be -1
+     */
+
+    expected = 1;
+    obtained = DirectFunctionCall2(int4_fixeddecimal_cmp, val1, val2);   
+
+    snprintf(expected_str, MAX_TEST_MESSAGE_LENGTH, "%ld", expected);
+    snprintf(obtained_str, MAX_TEST_MESSAGE_LENGTH, "%ld", obtained);
+
+    TEST_ASSERT(expected == obtained, expected_str, obtained_str, testResult);
+    return testResult;
+}
+
+
+
+
+TestResult*
+test_fixeddecimalum(void)
+{
+    /*
+     * To establish the expected behavior of our code, it is crucial to test both positive and negative scenarios.
+     * So in this test, we expect an out of range error.
+     * When this error occurs as anticipated, the test is considered successful.
+     */
+
+    int64 arg1 = INT64_MIN;
+    ErrorData *errorData;
+    MemoryContext oldcontext;
+
+    TestResult* testResult = palloc0(sizeof(TestResult));
+    testResult->result = false;
+
+    oldcontext = CurrentMemoryContext;
+    PG_TRY();
+    {
+        DirectFunctionCall1(fixeddecimalum, arg1);
+    }
+    PG_CATCH();
+    {
+        MemoryContextSwitchTo(oldcontext);
+        errorData = CopyErrorData();
+        FlushErrorState();
+        strncpy(testResult->message, errorData->message, MAX_TEST_MESSAGE_LENGTH);
+        testResult->result = true;
+        FreeErrorData(errorData);
+    }
+    PG_END_TRY();
+
+    return testResult;
```

Comment:
what is the message in case error does not happen?

Reply:
I missed this thing
So, now I added 
if(testResult->result == false)
```suggestion
    return testResult;
```
        strncpy(testResult->message, ", Out of Range error doesn't occur", MAX_TEST_MESSAGE_LENGTH);